### ISSUE TITLE
Issue with server side rendering

### ISSUE THREAD

**Issue by Nico924** [Mastermindzh/react-cookie-consent#43]
Hello,

I have a small issue in my server side rendered react app.

The config I used is very close to the one that can be found here :
https://github.com/manuelbieh/react-ssr-setup

The behaviour I have in production mode is : 

The cookie banner appears when loaded from the server (the cookie CookieConsent variable is not being used I guess)

Once the app is hydrated the cookie banner disappear. (Now the cookie is read) 

Does someone has experienced this issue before or have any advice to avoid that ?

Thank you in advance.

**Comment by Mastermindzh** [comment]
Can you clarify what is happening please?
It "just" works for me when I do:

1. git clone git@github.com:manuelbieh/react-ssr-setup.git
2. npm install
3. npm install --save react-cookie-consent
4. edit "src/shared/App.ts" with the following code:

![imgur picture of my code changes](https://i.imgur.com/6KYkuJP.png)

5. npm start


Things I've tried:

- Click accept
- Check cookie (exists)
- refresh (no bar)
- force refresh (no bar)
- clear cache and refresh (no bar)
- emulate slow mobile network (in chromium)
- quit and restart service.


Please give me an example project where the bar doesn't work as expected.

**Comment by Nico924** [comment]
Hello @Mastermindzh,

Thank you for your quick reaction.

I just understood where my "bug" came from. 
In the end of the **build** script of react-ssr-setup, there is a **generateStaticHTML** method which generates an **index.html** of the first page of the underlying web app in case of you want to host only a static frontend website. (this index file was along the other js/css file needed by the app).

Therefore when going on the first page of the app, the cookie part was present in the html while the JS was loading causing the quick apparition of the cookie banner.

Bottom line I just edited the build script in order to not have the static index.html generated if not wanted.

**Comment by Mastermindzh** [comment]
@Nico924 Cool!

Glad you figured it out and good luck with your project!
Don't forget to star the repo if you like it :D
